node-wasm/src/lock.rs

```diff
@@ -0,0 +1,186 @@
+// TODO: upstream to `named_lock`
+
+use std::fmt;
+
+use js_sys::{Function, Object, Promise, Reflect};
+use lumina_utils::make_object;
+use serde_wasm_bindgen::to_value;
+use tokio::sync::oneshot;
+use wasm_bindgen::prelude::*;
+use wasm_bindgen_futures::future_to_promise;
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error("Call would block")]
+    WouldBlock,
+
+    #[error("Could not acquire lock manager")]
+    LockManagerUnavailable(JsError),
+}
+
+pub struct NamedLock {
```

Comment:
I can't see drop impl or any other unlocking, am I missing sth?

Reply:
oh, actually we unlock on drop, but returning an error on receiver side. I mean, it works :shrug: but it isn't very obvious